[e2e tests] README.md - Add link to WooCommerce Monorepo prerequisites

alvarothomas opened the pull request:
Add a link to the WooCommerce Monorepo prerequisites before going through the e2e-pw prerequisites.

### Changes proposed in this Pull Request:

Following the WooCommerce E2E environment set up steps could lead to an error by not having some of the dependencies required to build the WooCommerce project.

Specifically, Composer is not included among the prerequisites to set up WooCommerce E2E Playwright environment, which likewise requires PHP. For that reason, installing the dependencies led to an error due to not not having Composer installed. Going to the steps described in the WooCommerce Monorepo prerequisites is needed beforehand.

This PR aims to include a link to those [WooCommerce Monorepo prerequisites](https://github.com/woocommerce/woocommerce/blob/trunk/README.md#prerequisites), while avoiding duplicated content by using documentation which is already available.

For additional context, please refer to [this Slack thread](https://a8c.slack.com/archives/C01BWDDTGKX/p1704893298832459).

### How to test the changes in this Pull Request:

The only way to reproduce the issue 

Run a clean installation to set up WooCommerce Playwright End to End Tests following the updated documentation.

1. PRE: WooCommerce project and its dependencies are not available
2. Follow [WooCommerce Playwright End to End Tests introduction](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/tests/e2e-pw#introduction)
3. Notice the results running `pnpm install`, which should install required dependencies
4. No error message should show up

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message


#### Comment 

Update E2E Playwright environment README.md file with a link to the WooCommerce Monorepo prerequisites to avoid duplicate content.

</details>

alvarothomas committed:
Update README.md

Add a link to the WooCommerce Monorepo prerequisites before going through the e2e-pw prerequisites.

Stojdza commented:
The change looks good (I followed that setup when I was setting up WC and e2e tests locally), but I couldn't figure out why the "Run lint checks potentially affecting projects across the monorepo" job is failing, so I will let someone else to review the PR and I'll 👀

adimoldovan commented on plugins/woocommerce/tests/e2e-pw/README.md:
```diff
@@ -19,6 +19,8 @@ This is the documentation for the new e2e testing setup based on Playwright and
 
 ## Pre-requisites
 
+In case this is the first time building WooCommerce, it is advised to go through the [WooCommerce Monorepo prerequisites]([url](https://github.com/woocommerce/woocommerce/blob/trunk/README.md#prerequisites)) first.
```
The URL is not rendered in GitHub, probably because the's an extra [url] in there between the parentheses.

alvarothomas replied:
Nice catch, I'm updating this. Thank you!

adimoldovan commented on plugins/woocommerce/tests/e2e-pw/README.md:
```diff
@@ -19,6 +19,8 @@ This is the documentation for the new e2e testing setup based on Playwright and
 
 ## Pre-requisites
 
+In case this is the first time building WooCommerce, it is advised to go through the [WooCommerce Monorepo prerequisites]([url](https://github.com/woocommerce/woocommerce/blob/trunk/README.md#prerequisites)) first.
```
Just a nitpick, but it's usually better to use relative links to files in the same repository, it will be less likely to break.

adimoldovan reviewed (commented):
Thanks for updating the docs @alvarothomas!

I left two comments below, feel free to ignore the nitpick. 

The build is failing because you'll need to also add a changelog file, describing your contribution. The error in the failing job mentions how to add the changelog. See: https://github.com/woocommerce/woocommerce/actions/runs/7501844141/job/20443677403?pr=43552#step:4:15

Commit:
Update WC prerequisites to relative link and added changelog

Update link to WC monorepo prerequisites

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

alvarothomas commented:
❓ Does it make sense to consider removing [these lines](https://github.com/woocommerce/woocommerce/pull/43552/files#diff-7278e11f22f3f5e1233ee9eb52d3f04c038d89729ec989fd7ef38120cbe1374eR24-R27) too?

Only installing Docker would be missing from there, if I'm not mistaken. We could replace all prerequisites with that link to the WC monorepo readme, and only add Docker installation as a reference.

Stojdza commented:
> ❓ Does it make sense to consider removing [these lines](https://github.com/woocommerce/woocommerce/pull/43552/files#diff-7278e11f22f3f5e1233ee9eb52d3f04c038d89729ec989fd7ef38120cbe1374eR24-R27) too?
> 
> Only installing Docker would be missing from there, if I'm not mistaken. We could replace all prerequisites with that link to the WC monorepo readme, and only add Docker installation as a reference.

This makes sense to me 👍 Follow WC prerequisites and additionally install Docker 🎯

Stojdza commented on plugins/woocommerce/tests/e2e-pw/README.md:
```diff
@@ -19,6 +19,8 @@ This is the documentation for the new e2e testing setup based on Playwright and
 
 ## Pre-requisites
 
+In case this is the first time building WooCommerce, it is advised to go through the [WooCommerce Monorepo prerequisites](https://github.com/woocommerce/woocommerce/blob/trunk/README.md#prerequisites) first.
```
I think what @adimoldovan meant in [his comment](https://github.com/woocommerce/woocommerce/pull/43552#discussion_r1452270753) about relative path is to avoid using absolute path with `https://github.com/...` and instead point to the .md file with something like `../../../../README.md#prerequisites`. More info about relative paths [here](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-readmes#:~:text=Relative%20links%20are%20easier%20for%20users%20who%20clone%20your%20repository.%20Absolute%20links%20may%20not%20work%20in%20clones%20of%20your%20repository%20%2D%20we%20recommend%20using%20relative%20links%20to%20refer%20to%20other%20files%20within%20your%20repository.).  Using relative paths also means that documentation can stand on its own without referencing, in our case, to GitHub. 

Since we're using an absolute path in the rest of this README file, I would say it's okay to leave it now with `https://github.com/...` and change all paths as a part of separate PR in the future.

nigeljamesstevenson replied:
I like the idea of relative paths based on the link Ivan shared. @Stojdza - are you ok to create an Issue in our [quality repo](https://github.com/woocommerce/woocommerce-quality/issues) to update this? 🙂

nigeljamesstevenson replied:
gh-woocommerce/woocommerce-quality

Stojdza replied:
Done - https://github.com/woocommerce/woocommerce-quality/issues/761

adimoldovan committed:
Merge branch 'trunk' into README.md]-Add-link-to-WC-Monorepo-prerequisites

adimoldovan committed:
Merge branch 'trunk' into README.md]-Add-link-to-WC-Monorepo-prerequisites

adimoldovan commented:
@alvarothomas are you still planning on updating the docs further as discussed in https://github.com/woocommerce/woocommerce/pull/43552#issuecomment-1893525608 or should we merged this one and get to that later in another PR?

Commit:
Remove duplicated lines

Remove duplicated instructions and leave the step to install Docker and Docker compose

alvarothomas commented:
Done, thanks for the heads-up @adimoldovan . We can add the relative links in another PR.

adimoldovan reviewed (approved):
LGTM!
